### ISSUE TITLE
Single arm model performance is worse on pi0.5 when compared with pi0

### ISSUE THREAD

**Issue by perseusdg** [Physical-Intelligence/openpi#692]
Hello, 

I have finetuned pi0 for a single Trossen Widowx AI arm ( I believe this might be the successor to the Widowx 250). I modified the aloha policy for a single arm (7DOF) and kept the rest largely unchanged. My dataset consists of simple pick of a red cube for about 300 trajectories within a 6X12 area in front of it.  I finetuned both models using lora (150,000+ timesteps for pi0.5 and about 80,000+ timesteps for pi0) both were trained on 4xA100 (40GB) . The only major difference between them was I used fsdp=4 for pi0 while for pi0.5 I used the default (fsdp=1) . 

In terms of the actual behavior when I rollout the policy pi0.5 isn't able to grasp the red cube at all points within the 6x12 area, while pi0 has a near 100 percent success rate. Additionally pi0 was able to do pick and place tasks  to some extent even though the dataset I trained it with had explicitly pick tasks only, while pi0.5 sort of hallucinates and reaches for the table. 

Should I not use lora or freeze filters for pi0.5 ? or is there something I am doing wrong in terms of the actual config? 

Pi0 Config : 
```
TrainConfig(
    name="pi0_original_widowx_solo_finetune",
        model=pi0.Pi0Config(
            pi05=False,
            action_horizon=50,
            paligemma_variant="gemma_2b_lora",
            action_expert_variant="gemma_300m_lora"
        ),
        data=LeRobotWidowxSoloDataConfig(
            repo_id = "pick_6x12_300",  
            adapt_to_pi=False,
            use_delta_joint_actions=False,
            base_config=DataConfig(
                prompt_from_task=False
            ),
            default_prompt="Pick up red cube",
            repack_transforms=_transforms.Group(
                inputs=[
                    _transforms.RepackTransform(
                        {
                            "images":{
                                "cam_main" : "observation.images.cam_main",
                                "cam_wrist" : "observation.images.cam_wrist"
                            },
                            "state" : "observation.state",
                            "actions" : "action",
                        }
                    )
                ]
            ),
        ),
        weight_loader=weight_loaders.CheckpointWeightLoader("gs://openpi-assets/checkpoints/pi0_base/params"),
        num_train_steps=100_000,
        freeze_filter=pi0.Pi0Config(
            pi05=False,
            action_dim=32, 
            action_horizon=50, 
            paligemma_variant="gemma_2b_lora",
            action_expert_variant="gemma_300m_lora"
        ).get_freeze_filter(),
        ema_decay=None
    )
```

Pi0.5 Config: 
 ```
TrainConfig(
    name="pi05_original_widowx_solo_finetune",
        model=pi0.Pi0Config(
            pi05=True,
            action_horizon=50,
            paligemma_variant="gemma_2b_lora",
            action_expert_variant="gemma_300m_lora",
        ),
        data=LeRobotWidowxSoloDataConfig(
            repo_id = "pick_6x12_300",  
            adapt_to_pi=False,
            use_delta_joint_actions=False,
            base_config=DataConfig(
                prompt_from_task=False
            ),
            default_prompt="Pick up the red cube",
            repack_transforms=_transforms.Group(
                inputs=[
                    _transforms.RepackTransform(
                        {
                            "images":{
                                "cam_main" : "observation.images.cam_main",
                                "cam_wrist" : "observation.images.cam_wrist"
                            },
                            "state" : "observation.state",
                            "actions" : "action",
                        }
                    )
                ]
            ),
        ),
        weight_loader=weight_loaders.CheckpointWeightLoader("gs://openpi-assets/checkpoints/pi05_base/params"),
        num_train_steps=200_000,
        freeze_filter=pi0.Pi0Config(
            pi05=True,
            action_dim=32, 
            action_horizon=50, 
            paligemma_variant="gemma_2b_lora",
            action_expert_variant="gemma_300m_lora"
        ).get_freeze_filter(),
        ema_decay=None
    )
```

Training Loss for Pi0.5 : 

<img width="1414" height="573" alt="Image" src="https://github.com/user-attachments/assets/85c6d8cb-015e-4d42-918e-e085f9ce7b7c" />

Training Loss for Pi0:

<img width="1411" height="564" alt="Image" src="https://github.com/user-attachments/assets/74f09faa-be0c-4e64-963e-3895a97355c1" />
 (even though the graph says 4xa40, I trained it on 4xa100 😅)

**Comment by cyteena** [comment]
dude, why you train steps for pi05 doubles the train steps for pi0? I think finetune steps is the key here? Have check the performance of middle ckpt ?

**Comment by perseusdg** [comment]
I saved checkpoints every 5000 steps and checked them from checkpoint_15000 to checkpoint_120000 and the performance was pretty bad across the board when compared to pi0.  

Yeah I agree it seems like fine-tuning is the key here but I am not sure whats wrong with my current config setup.

**Comment by cyteena** [comment]
pretty interesting. I think if it works for pi0, it should also work for pi05, it seems no big difference.

**Comment by perseusdg** [comment]
unfortunately not yet , but I will try it without lora and see how it goes . Do you suppose a batch size lower than 64 for pi0.5 impacts its performance maybe ?

**Comment by perseusdg** [comment]
I didn't get enough compute yet to train it without lora , but I tried again on a longer horizon task (clean the table and put all the items in a drawer) , pi0 exhibited behaviors of learning at 18,000 steps itself it was able to open/close the drawer while pi0.5 always fail to grasp the drawer and just ends up hallucinating. 

my batch size and action horizon are 32 and 50 respectively  for both pi0 and pi0.5 and I used lora for both without any EMA. I am absolutely baffled about this ,to me it doesn't make sense pi0 works but pi0.5 doesn't work.

**Comment by emoPointer** [comment]
I think you can close the **q01** and **q99** and try again with PI05.

**Comment by siyuhsu** [comment]
We have the same issue when fine-tuning pi0.5 on a single piper arm. The performance is noticeably worse compared to pi0 under similar settings.

Have you found any solution or insights since then?

**Comment by zwbx** [comment]
> I think you can close the **q01** and **q99** and try again with PI05.

Hey, what is the secret hidden here.

**Comment by Rseelan** [comment]
Have the same issue for single arm on pi05 as well

**Comment by anlorla** [comment]
> > I think you can close the **q01** and **q99** and try again with PI05.
> 
> Hey, what is the secret hidden here.

i think it's about normalization
